Source/WebKit/Platform/IPC/Connection.h

```diff
@@ -674,6 +662,49 @@ inline std::unique_ptr<Decoder> Connection::waitForMessageForTesting(MessageName
 }
 #endif
 
+template<typename T, typename C>
+Connection::AsyncReplyHandler Connection::makeAsyncReplyHandler(C&& completionHandler, ThreadLikeAssertion callThread)
+{
+    // FIXME: callThread by default uses AnyThread because the API contract on invalid sends does not make sense.
+    return AsyncReplyHandler {
+        {
+            [completionHandler = WTFMove(completionHandler)] (Decoder* decoder) mutable {
+                if (decoder && decoder->isValid())
+                    callReply<T>(*decoder, WTFMove(completionHandler));
+                else
+                    cancelReply<T>(WTFMove(completionHandler));
+            }, callThread
+        },
+        nextAsyncReplyHandlerID()
+    };
+}
+
+template<typename T, typename C>
+void Connection::callReply(Decoder& decoder, C&& completionHandler)
+{
+    if constexpr (!std::tuple_size_v<typename T::ReplyArguments>) {
+        // Nothing to decode in case of no reply arguments, so just invoke the completion handler in that case.
+        completionHandler();
+    } else {
+        auto replyArguments = decoder.decode<typename T::ReplyArguments>();
+        if (!!replyArguments) {
```

Comment:
Is the `!!` required here? We Rey to avoid that in WebKit code. I suggest:
```
if (auto arguments = decoder.decode<typename T::ReplyArguments>()) {
```

Reply:
Adopted.